run.py

```diff
@@ -41,6 +40,14 @@ def run_step(self, name: str, script_path: str, timeout: int = 10 * 60) -> StepR
         print(stderr)
         print(stdout)
 
+        if not get_durtion_from_stdout:
+            elapsed = time.time()
+            duration = round(elapsed - start, 3)
+        else:
+            try:
+                duration = float(stdout)
```

Comment:
The implementation is risky as there might be some other stuff on stdout as well. 
Let's do it this way:
Expect there would a line in the stdout with this format: 
`::::DURATION=value` 
find the latest occurrence of it and extract the `value` out of it.

Reply:
Yes, I totally agree ... but what about a JSON output?

Reply:
I can't think of any situation we would need more than one key-value per line. 
+ generating JSON in the shell is quite counterintuitive.